lib/pure/collections/sequtils.nim

```diff
@@ -169,6 +169,24 @@ func cycle*[T](s: openArray[T], n: Natural): seq[T] =
       result[o] = e
       inc o
 
+func upsample*[T](s: openArray[T], n: Natural): seq[T] =
```

Comment:
I'm not sure how useful this is.  In what scenarios would this kind of function be needed?

Reply:
[Upsampling](https://en.wikipedia.org/wiki/Upsampling) is quite an important function in dsp.